### PR TITLE
Add a not found error to admission control

### LINKED ISSUE #14011: exists admission controller should return 404, not 403.
Currently the `exists` admission controller returns a forbidden (403) error if the namespace doesn't exist.

We should return a 404 error instead.

### PR DESCRIPTION
**Pull request by brendandburns**
@bgrant0607 @lavalamp 

This closes https://github.com/kubernetes/kubernetes/issues/14011

### CODE REVIEW

**Comment by lavalamp** [comment]
LGTM

File: pkg/admission/errors.go [hunk 1]
```diff
@@ -46,3 +47,21 @@ func NewForbidden(a Attributes, internalError error) error {
 	}
 	return apierrors.NewForbidden(kind, name, internalError)
 }
+
+func NewNotFound(a Attributes) error {
```
Comment (lavalamp) [inline pkg/admission/errors.go]: A godoc line would be nice :)

**Comment by bgrant0607** [comment]
cc @derekwaynecarr @nikhiljindal

**Comment by derekwaynecarr** [comment]
@brendandburns - we merged ```NamespaceExists``` and ```NamespaceLifecycle``` post 1.0.  We need to make the same change to ```NamespaceLifecycle``` here: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/namespace/lifecycle/admission.go#L85

**Comment by derekwaynecarr** [comment]
Removed the LGTM because the change needs to also happen in another admission handler for master branch.

File: pkg/admission/errors.go (hunk 1 above)
Comment (derekwaynecarr) [inline pkg/admission/errors.go]: There is a lot of common code between this and NewForbidden to determine the name for a resource.  Can you make a utility and share it between the functions?

Comment (brendandburns) [inline pkg/admission/errors.go]: done and done.

**Comment by brendandburns** [comment]
@derekwaynecarr comments addressed, please re-check.

Thanks!
--brendan

**Comment by derekwaynecarr** [comment]
LGTM

**Comment by alex-mohr** [comment]
@k8s-bot test this again

**Pull request merged**